integrationtests/controller/bundle/bundle_targets_test.go

```diff
@@ -0,0 +1,369 @@
+package agent
+
+import (
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/rancher/fleet/integrationtests/utils"
+	"github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+	v1gen "github.com/rancher/fleet/pkg/generated/controllers/fleet.cattle.io/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+var (
+	env                        *specEnv
+	bundleController           v1gen.BundleController
+	clusterController          v1gen.ClusterController
+	bundleDeploymentController v1gen.BundleDeploymentController
+	clusterGroupController     v1gen.ClusterGroupController
+)
+
+var _ = Describe("Bundle targets", Ordered, func() {
+	BeforeAll(func() {
+		env = specEnvs["targets"]
+		bundleController = env.fleet.V1alpha1().Bundle()
+		clusterController = env.fleet.V1alpha1().Cluster()
+		bundleDeploymentController = env.fleet.V1alpha1().BundleDeployment()
+		clusterGroupController = env.fleet.V1alpha1().ClusterGroup()
+
+		createClustersAndClusterGroups()
+
+		DeferCleanup(func() {
+			Expect(env.k8sClient.Delete(ctx, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: env.namespace}})).ToNot(HaveOccurred())
+		})
+	})
+
+	var (
+		targets            []v1alpha1.BundleTarget
+		targetRestrictions []v1alpha1.BundleTarget
+		bundleName         string
+		bdLabels           map[string]string
+	)
+
+	JustBeforeEach(func() {
+		bundle, err := createBundle(bundleName, env.namespace, bundleController, targets, targetRestrictions)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(bundle).To(Not(BeNil()))
+	})
+
+	AfterEach(func() {
+		Expect(bundleController.Delete(env.namespace, bundleName, nil)).NotTo(HaveOccurred())
+	})
+
+	When("target all clusters without customization", func() {
+		BeforeEach(func() {
+			bundleName = "all"
+			bdLabels = map[string]string{
+				"fleet.cattle.io/bundle-name":      bundleName,
+				"fleet.cattle.io/bundle-namespace": env.namespace,
+			}
+			// simulate targets in GitRepo. All targets in GitRepo are also added to targetRestrictions, which acts as a white list
+			targets = []v1alpha1.BundleTarget{
+				{
+					ClusterGroup: "all",
+				},
+			}
+			targetRestrictions = make([]v1alpha1.BundleTarget, len(targets))
+			copy(targetRestrictions, targets)
+		})
+
+		It("three BundleDeployments are created", func() {
+			var bdList = verifyBundlesDeploymentsAreCreated(3, bdLabels)
```

Comment:
we don't always pass 3. For example [here](https://github.com/raulcabello/fleet/blob/b338a55e227d08ef48f0d37605eeacafe45b5a8f/integrationtests/controller/bundle/bundle_targets_test.go#L272). In the future we might end up with many different numbers, and in this case I think it is more readable like it currently is.

Reply:
Sure, what I feel is important here though is _what_ 3 means here, namely the number of clusters that we have. We could have a constant be `var clusterNames = []string{"one", "two", "three"}`, pass that to `createClustersAndClusterGroups`, and use `verifyBundlesDeploymentsAreCreated(len(clusterNames), bdLabels)` in cases where we currently use 3 instead.
I feel that this may prevent future readers from wondering where 3 comes from.
How much sense would that make to you?

Reply:
I don't think using` len(clusterNames)` is a good idea, because sometimes we want to assert that not all BundleDeployments were created. I extracted it to a var which I think makes it clearer in https://github.com/raulcabello/fleet/commit/f9457f9179f7922236bcaf979c4a7b4ced18133e 
let me know what you think

Reply:
That also works for me, as it clarifies the meaning of those numbers :) Thanks!